src/transformers/pipelines/__init__.py

```diff
@@ -727,28 +733,48 @@ def pipeline(
     if isinstance(model, Path):
         model = str(model)
 
+    if hub_kwargs["_commit_hash"] is None:
+        pretrained_model_name_or_path = None
+        if isinstance(config, str):
+            pretrained_model_name_or_path = config
+        elif config is None and isinstance(model, str):
+            pretrained_model_name_or_path = model
+
+        if not isinstance(config, PretrainedConfig) and pretrained_model_name_or_path is not None:
+            # We make a call to the config file first (which may be absent) to get the commit hash as soon as possible
+            resolved_config_file = cached_file(
+                pretrained_model_name_or_path,
+                CONFIG_NAME,
+                _raise_exceptions_for_missing_entries=False,
+                _raise_exceptions_for_connection_errors=False,
+                **hub_kwargs,
+            )
+            hub_kwargs["_commit_hash"] = extract_commit_hash(resolved_config_file, commit_hash)
+        else:
+            hub_kwargs["_commit_hash"] = getattr(config, "_commit_hash", None)
+
     # Config is the primordial information item.
     # Instantiate config if needed
     if isinstance(config, str):
-        config = AutoConfig.from_pretrained(config, _from_pipeline=task, **hub_kwargs, **model_kwargs)
+        config = AutoConfig.from_pretrained(
+            config, _from_pipeline=task, code_revision=code_revision, **hub_kwargs, **model_kwargs
+        )
         hub_kwargs["_commit_hash"] = config._commit_hash
     elif config is None and isinstance(model, str):
         # Check for an adapter file in the model path if PEFT is available
         if is_peft_available():
-            subfolder = hub_kwargs.get("subfolder", None)
-            maybe_adapter_path = find_adapter_config_file(
-                model,
-                revision=revision,
-                token=use_auth_token,
-                subfolder=subfolder,
-            )
+            # `find_adapter_config_file` doesn't accept `trust_remote_code`
+            _hub_kwargs = {k: v for k, v in hub_kwargs.items() if k != "trust_remote_code"}
```

Comment:
Can we just pass explicitly the arguments as it was done before? This seems a bit brittle

Reply:
Look this part 

https://github.com/huggingface/transformers/blob/0ac3875011d32dc85e0e83970507e3afe8f0febb/src/transformers/modeling_utils.py#L2511-L2522 (this [diff in modeling_utils.py](https://github.com/huggingface/transformers/commit/2febd506149d039b51590f5dc7b45f0d8624819d) might be more clear)

it passes more arguments to `find_adapter_config_file`. I can definitely use (a long list of) explicitly arguments here.